src/Features/Core/Portable/Completion/Providers/AbstractObjectInitializerCompletionProvider.cs

```diff
@@ -96,5 +88,42 @@ protected virtual bool IsInitializable(ISymbol member, INamedTypeSymbol containi
                 member.MatchesKind(SymbolKind.Field, SymbolKind.Property) &&
                 member.IsAccessibleWithin(containingType);
         }
+
+        private static bool CanSupportObjectInitializer(ISymbol symbol, ISymbol within)
+        {
+            if (symbol is IPropertySymbol propertySymbol)
+            {
+                return propertySymbol.GetMethod != null && !propertySymbol.Type.IsStructType() && HasAccesseblePropertiesOrFields(propertySymbol.Type, within);
+            }
+
+            return false;
+        }
+
+        private static bool HasAccesseblePropertiesOrFields(ITypeSymbol type, ISymbol within)
```

Comment:
Typo: Accessible

Reply:
📝 This test can be removed altogether. Empty braces are allowed for the initializer even if no members are accessible.

Reply:
@sharwell so we'll show the suggestion for all valid syntax (even if no settable properties/fields)?